### DISCUSSION TITLE
Preference Modification on Workspace Level

### DISCUSSION THREAD

**Discussion by maheshrajrp** [eclipse-theia/theia#10118] [General] [answered]
Hi,
I'm having a dialog box that can change preference. However, it changes preference of "User Preference", but I wish to make those those changes on Workspace level/ "Workspace Preferences". Any examples or documents regarding it would be greatly helpful.

Thanks

**Comment by msujew** [comment, marked as answer]
Hi @maheshrajrp,

Assuming you are using the `PreferenceService` to modify the preference, you can use the [`set`](https://github.com/eclipse-theia/theia/blob/0221c6179f7ea16673861fa1f602868a1f54d986/packages/core/src/browser/preferences/preference-service.ts#L160) method with the `PreferenceScope.Workspace` enum value as its third argument.

**Reply by maheshrajrp** [reply]
Thanks. This should do the trick !!
